theseus/core/robust_cost_function.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Tuple
+
+import torch
+
+import theseus as th
+
+from .loss import Loss
+
+
+class RobustCostFunction(th.CostFunction):
+    def __init__(
+        self,
+        loss_function: Loss,
+        log_loss_radius: th.Vector,
+        cost_function: th.CostFunction,
```

Comment:
Is the idea of this implementation that any cost function can be turned into its robust version?

Reply:
Yes.